src/deque.jl

```diff
@@ -195,4 +195,4 @@ julia> setindex(@SMatrix[2 4; 6 8], 1, 2)
 end
 
 # TODO proper multidimension boundscheck
-@propagate_inbounds setindex(a::StaticArray, x, inds::Int...) = setindex(a, x, LinearIndices(a)[inds...])
+@propagate_inbounds setindex(a::StaticArray, x, inds::Union{Integer, CartesianIndex}...) = setindex(a, x, LinearIndices(a)[inds...])
```

Comment:
I'm new to StaticArrays internal so I might be wrong.
Is there any reason why we don't use `to_indices` here?

```julia
@propagate_inbounds setindex(a::StaticArray, x, inds...) = setindex(a, x, LinearIndices(a)[to_indices(a, inds)...])
```

This supports the hybrid of many cases, e.g., `setindex(a, x, 1, CartesianIndex(2,))`.

Reply:
I'm trying this.

Reply:
What is the advantage of inserting `to_indices`? Won't
`LinearIndices(a)[inds...]`
call `to_indices(LinearIndices(a), inds)` anyway?

Reply:
Ooh you're right -- my first draft version doesn't use `LinearIndices` and it triggers recursive self-calling and OverflowError so I blindly added back the `LinearIndices` 🙈

Reply:
Yes, we should remove the `::Union{Integer, CartesianIndex}` annotation.

Reply:
I updated the code. Please have a look at the current version
```Julia
@propagate_inbounds setindex(a::StaticArray, x, inds...) = setindex(a, x, LinearIndices(a)[inds...])
```
and tell me if that should be improved.

Reply:
I think that looks good.